src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name);
+    this.health = health;
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+
+    return this.hidden;
+  }
+
+  apply(sourceHealth, damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name);
```

Comment:
Similar to the `Herbivore` class, the `super(name)` call should also pass the `health` parameter to the parent `Animal` class constructor. It should be `super(name, health);`.